Firestore/core/test/unit/model/value_util_test.cc

```diff
@@ -132,6 +133,15 @@ class ValueUtilTest : public ::testing::Test {
     EXPECT_EQ(expected_canonical_id, actual_canonical_id);
   }
 
+  void VerifyDeepClone(const google_firestore_v1_Value& value) {
+    // `Message` takes ownership of the proto value and destroys it. We use
```

Comment:
Hmm, I'm not sure if it verifies that values own their own memory -- `clone1` and `clone2` will be valid for the duration of the test. If I understand the intention correctly, consider something like:
```cpp
    nanopb::Message<google_firestore_v1_Value> clone1;
    [&] {
      nanopb::Message<google_firestore_v1_Value> clone2{DeepClone(value)};
      EXPECT_TRUE(value == *clone2);
      clone1 = nanopb::Message<google_firestore_v1_Value>{DeepClone(*clone2)};
    }();
    // clone2 is destroyed at this point, but clone1 should be still valid.
    EXPECT_TRUE(value == *clone1);
```

Reply:
The idea was more to cause a segfault since both messages would free the same memory regions. Your code snippet is better though as it would catch these errors during runtime.